rero_ils/modules/documents/serializers.py

```diff
@@ -74,15 +74,14 @@ def preprocess_record(self, pid, record, links_factory=None, **kwargs):
     def post_process_serialize_search(self, results, pid_fetcher):
         """Post process the search results."""
         # Item filters.
-        viewcode = request.args.get('view',  current_app.config.get(
-                'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'
-        ))
+        global_view_code = current_app.config.get(
+            'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE')
+        viewcode = request.args.get('view', global_view_code)
         records = results.get('hits', {}).get('hits', {})
         for record in records:
             metadata = record.get('metadata', {})
-            available = Document.get_record_by_pid(
-                metadata.get('pid')).is_available(viewcode)
-            metadata['available'] = available
+            document = Document.get_record_by_pid(metadata.get('pid'))
+            metadata['available'] = document.is_available(viewcode)
```

Comment:
This two lines are responsible for a lot of delay during display of the results. We have to find a faster or asynchron way to do it.

Reply:
You're right, potentially we read all holdings and all items to check availability.
But, if we found one related available holding/item, the loops breaks.
Don't' know how to improve this code, because it's needed by UI at this time.

I think it should be managed into a separate PR/issue (because it will have a huge impact on UI)